src/hotspot/cpu/x86/x86.ad

```diff
@@ -2107,6 +2109,17 @@ const bool Matcher::match_rule_supported_vector_masked(int opcode, int vlen, Bas
       if ((bt == T_INT || bt == T_LONG) && VM_Version::supports_avx512cd()) {
         return true;
       }
+    case Op_AddVHF:
+    case Op_SubVHF:
+    case Op_MulVHF:
+    case Op_DivVHF:
+    case Op_AbsVHF:
+    case Op_NegVHF:
+      if(bt != T_SHORT && !VM_Version::supports_avx512_fp16()) {
+        return false;
+      }
```

Comment:
Should it be moved Matcher::match_rule_supported_vector, backend does not support predicated half floats instructions currently.